### PR TITLE
search_book_website

### PR DESCRIPTION
**Pull request by VanHoa18042000**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/51511)

## WHAT (optional)
- Add theme and handle for search books ajax

## HOW
- Add css, js for home page , admin , layout login . And add gem i18n

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
https://edu-redmine.sun-asterisk.vn/issues/51511

### CODE REVIEW

**Comment by VanHoa18042000** [comment]
image

**Comment by VanHoa18042000** [comment]
ready

File: app/controllers/website/search_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Website::SearchController < WebsiteController
+  def index
+    @books = Book.search(params)
```
Comment (huongnt-2545) [inline app/controllers/website/search_controller.rb:3]: thêm phân trang em nhé
Đang search theo những tiêu chí nào thế em?

File: app/helpers/website/search_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module Website::SearchHelper
```
Comment (huongnt-2545) [inline app/helpers/website/search_helper.rb:1]: ko cần thì xóa file đi em

File: app/models/book.rb [hunk 3]
```diff
@@ -21,4 +21,20 @@ class Book < ApplicationRecord
...
+    end
+  end)
+  scope :search_author, (lambda do |author|
+    where(author_id: author) if author.present?
+  end)
+  scope :search, (lambda do |params|
+    search_book_name(params[:book_name])
+      .search_author(params[:author])
+      .search_category(params[:category])
+  end)
```
Comment (huongnt-2545) [inline app/models/book.rb:39]: xử lý gọi bên controller đi em nhé
Ko nên define 1 scope để chỉ gọi các scope khác

File: app/views/website/search/index.html.erb [hunk 4]
```diff
@@ -12,33 +12,38 @@
             <hr>
           </div>
         </div>
-        <form class="row">
-          <div class="col-12 col-sm-12 col-md- col-lg- col-xl- col-xxl-4">
-            <input type="" name="" placeholder="Nhập tên sách các bạn muốn tìm ..." class="form-control">
+        <%= form_with url: search_products_path, method: :get, local: false do |f| %>
+          <div class="row">
+            <div class="col-12 col-sm-12 col-md- col-lg- col-xl- col-xxl-4">
+              <%= f.text_field :book_name, id: :book_name, placeholder: "Nhập tên sách các bạn muốn tìm ...", class:"form-control" %>
```
Comment (huongnt-2545) [inline app/views/website/search/index.html.erb:18]: I18n text

File: app/views/website/search/index.html.erb [hunk 5]
```diff
@@ -12,33 +12,38 @@
...
+            </div>
+            <div class="col-12 col-sm-12 col-md- col-lg- col-xl- col-xxl-3">
+              <%= f.select :category, options_for_select(Category.pluck(:name, :id)), {include_blank: t("choose_category")},
+              {:class => "form-select"} %>
```
Comment (huongnt-2545) [inline app/views/website/search/index.html.erb:22]: sai indent

**Comment by VanHoa18042000** [comment]
ready

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**